pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go

```diff
@@ -534,3 +535,16 @@ func pvPhaseTransformer(s sender.Sender, name string, metric ksmstore.DDMetric,
 func serviceTypeTransformer(s sender.Sender, name string, metric ksmstore.DDMetric, hostname string, tags []string, _ time.Time) {
 	submitActiveMetric(s, ksmMetricPrefix+"service.type", metric, hostname, tags)
 }
+
+// removeSecretTransformer removes the secret tag from the kube_ingress_tls metric
+func removeSecretTransformer(s sender.Sender, _ string, metric ksmstore.DDMetric, hostname string, tags []string, _ time.Time) {
+	if len(tags) > 0 {
+			if strings.HasPrefix(s, "secret:") {
+				tags = lo.Filter(tags, func(x string, index int) bool {
+					return x != s
+				})
+			}
+		}
```

Comment:
what I was think was the opposite

 ```suggestion
         tags = lo.Filter(tags, func(x string, _ int) bool { return !strings.HasPrefix(x, "secret:") })
```